src/ORM/FieldType/DBYear.php

```diff
@@ -25,22 +36,62 @@ public function scaffoldFormField(?string $title = null, array $params = []): ?F
         return $selectBox;
     }
 
+    public function setValue(mixed $value, null|array|ModelData $record = null, bool $markChanged = true): static
```

Comment:
It looks like ALL of the "niceness" in here is specific to `MySQL` which isn't appropriate.

Reply:
We could just rename this class to DBMySQLYear if you want, and keep the 'Year' key in models.yml so things still work as expected

That said, I'd rather just keep the class name is and maybe put in a docblock saying this is only meant for MySQL databases (which is the only database we support so :shrug:)